pkg/resolution/resolver/git/git.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package git
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+type repository struct {
+	url       string
+	username  string
+	password  string
+	directory string
+	revision  string
+}
+
+func resolveRepository(ctx context.Context, url, username, password, revision string) (*repository, func(), error) {
```

Comment:
Maybe we can export the repository struct and use it in the function as an argument instead of passing four arguments ?

Reply:
I can rework this a bit to make it more ergonomic. I am hesitant to export this struct but I think there is a way to work around that

Reply:
LMK what you think about the refactor

Reply:
Maybe something for a follow-up?